Header.h

```diff
@@ -23,7 +23,7 @@ typedef struct Header_ {
    int height;
 } Header;
 
-#define Header_forEachColumn(this_, i_) for (size_t (i_)=0; (i_) < HeaderLayout_getColumns((this_)->headerLayout); ++(i_))
+#define Header_forEachColumn(this_, i_) for (size_t (i_)=0, numColumns__ = HeaderLayout_getColumns((this_)->headerLayout); (i_) < numColumns__; ++(i_))
```

Comment:
Just a reminder: Double underscores are reserved in C++ standard. It's a good idea to avoid them in C program as well.
```suggestion
#define Header_forEachColumn(this_, i_) for (size_t (i_)=0, numColumns_ = HeaderLayout_getColumns((this_)->headerLayout); (i_) < numColumns_; ++(i_))
```

Reply:
AFAIR this was just for identifiers starting in double underscore …

Reply:
https://en.cppreference.com/w/cpp/language/identifiers:
> the identifiers with a double underscore anywhere are reserved

However C++ compatibility is generally broken, e.g by the usage of `this` as variable name.